ui/src/actions/remoteAccess.js

```diff
@@ -52,154 +42,64 @@ export function requestControl(
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      body: JSON.stringify({
-        control,
-        message,
-        name,
-        userInfo,
-      }),
-    });
+    sendRequestControl(control, message, name, userInfo);
   };
 }
 
-export function sendTakeControl() {
+export function takeControl() {
   return (dispatch) => {
-    fetch('mxcube/api/v0.1/ra/take_control', {
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-    }).then(() => {
+    sendTakeControl().then(() => {
       dispatch(getLoginInfo());
       dispatch(getRaState());
     });
   };
 }
 
-export function sendGiveControl(username) {
+export function giveControl(username) {
   return (dispatch) => {
-    fetch('mxcube/api/v0.1/ra/give_control', {
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      body: JSON.stringify({ username }),
-    }).then(() => {
+    sendGiveControl(username).then(() => {
       dispatch(getLoginInfo());
       dispatch(getRaState());
     });
   };
 }
 
-export function sendLogoutUser(username) {
+export function logoutUser(username) {
   return (dispatch) => {
-    fetch('mxcube/api/v0.1/ra/logout_user', {
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      body: JSON.stringify({ username }),
-    }).then(() => {
+    sendLogoutUser(username).then(() => {
       dispatch(getLoginInfo());
       dispatch(getRaState());
     });
   };
 }
 
-export function requestControlResponse(giveControl = true, message = '') {
-  fetch('mxcube/api/v0.1/ra/request_control_response', {
-    method: 'POST',
-    credentials: 'include',
-    headers: {
-      Accept: 'application/json',
-      'Content-type': 'application/json',
-    },
-    body: JSON.stringify({ giveControl, message }),
-  });
-
-  return { type: 'REQUEST_CONTROL_RESPONSE' };
-}
-
-export function setAllowRemoteAccess(allow) {
-  return { type: 'SET_ALLOW_REMOTE', allow };
-}
-
-export function setTimeoutGivesControl(timeoutGivesControl) {
-  return { type: 'SET_TIMEOUT_GIVES_CONTROL', timeoutGivesControl };
-}
-
-export function sendAllowRemote(allow) {
+export function respondToControlRequest(giveControl = true, message = '') {
   return (dispatch) => {
-    fetch('mxcube/api/v0.1/ra/allow_remote', {
-      method: 'POST',
-      credentials: 'include',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json',
-      },
-      body: JSON.stringify({ allow }),
+    sendRespondToControlRequest(giveControl, message).then(() => {
+      dispatch(getLoginInfo());
+      dispatch(getRaState());
```

Comment:
Now dispatching these actions after responding to a control request, so the modal with accept/deny closes itself and the state gets updated right away.

Reply:
:+1:

Reply:
Thanks !